### DISCUSSION TITLE
SQLite

### DISCUSSION THREAD

**Discussion by zubricks** [payloadcms/payload#4189] [Roadmap]
We’ve built out the adapter pattern for SQLite and will be able to finish support for SQLite once Drizzle completes the work on their end. From there we’ll perform thorough testing before merging in to core.

**Comment by felstormrage** [comment]
Will this support Turso?

**Reply by nathanaelphilip** [reply]
This is the big brain move.

**Reply by Xadoy** [reply]
Or better: astro db

**Comment by oskbor** [comment]
Excited for sqlite support! Any update on when it might be released?

**Comment by h0byte** [comment]
Would also love an update on this, excited to use Payload with Turso!

**Comment by mrdankuta** [comment]
Hi! Any updates on this? I am eager to see this released.

**Comment by kimstewartca** [comment]
Hi @zubricks 
Just following up on your post from last year.
Would you happen to know if there are any updates on this?
Thank you!

**Comment by janus-reith** [comment]
It would be nice to get an idea of the actual steps involved — Would this just work out of the box using drizzle or are there additional changes necessary, are they any hard blockers where some Postgres specific features are required that don't work the same way with sqlite?

**Comment by zubricks** [comment]
Hey everyone—before we can make this happen we need to get postgres stable. We’re aiming to have postgres complete for 3.0 launch (early next week). From there we can abstract the core logic to make it reusable across other relational dbs including SQLite—likely a couple of weeks after 3.0 launch.

We’re aiming for feature parity with postgres and keeping as much code unified as possible through the adapter. Our expectation is that everything will be able to work the same way.

**Comment by ainsleyclark** [comment]
Is there any update on this feature?

**Comment by ainsleyclark** [comment]
Is there any update on this feature?

**Reply by DanRibbens** [reply]
> Is there any update on this feature?

I have it nearly done on the branch feat/db-sqlite.

**Reply by ainsleyclark** [reply]
Fantastic thank you @DanRibbens 

Is there compatibility with Turso?

**Comment by BohdanK-W32** [comment]
Are there any updates/ETA on that feature?
@DanRibbens

**Reply by funabab** [reply]
Good news👏 #7320 was recently merged.

**Comment by DanRibbens** [comment]
Today we published the first beta release with SQLite: @payloadcms/db-sqlite@v3.0.0-beta.72.

Please give it a shot and share your feedback.

Thanks!

**Reply by Montreal-Dev** [reply]
Time to break the feature ?! (Thank you) 🎊 ❤️

**Comment by leomp12** [comment]
Hey @DanRibbens , is it also intended to work with remote SQLite, such as Cloudflare D1 and Turso?

**Reply by kimstewartca** [reply]
I am very much looking forward to edge-based hosting on CloudFlare Pages with D1 (SQLite) and R3.

The only other Payload related thing that I can wish for is image upload to multiple directories.

Hopefully it's on the request queue and coming soon!

**Reply by BohdanK-W32** [reply]
@kimstewartca, totally agree with you, for now there are two blockers (as I see): sharp (requires multi threading, which is not supported on Cloudflare due to security reasons) and maybe another plugin/extension for D1 (because their DB is not the default SQLite).

But the sharp is the main blocker, it would be really nice to implement an alternative adapter for the external image optimization services (Cloudflare Image Optimization, etc.)

**Reply by fabianeisenhut** [reply]
Hi @BohdanK-W32, just stumbled across this thread. Are there any updates on whether Payload / Sharp support D1 now?

**Reply by BohdanK-W32** [reply]
@fabianeisenhut , I'm curious too!
P.S.: I'm not a Payload team member 😅

**Comment by ainsleyclark** [comment]
This has worked in Turso like a dream, thank you for making this happen!

**Reply by 6ui11em** [reply]
How did you connect it to Turso?

**Reply by 6ui11em** [reply]
Ok, I've seen how to do it, I just need to configure DATABASE_URI and DATABASE_AUTH_TOKEN.

**Reply by logemann** [reply]
you managed to run it also on Vercel or Netlify? Cant get libsql thing working on those serverless platforms.

**Comment by pmcb99** [comment]
when I create a new project selecting website and sqlite, the tables get made but something is messed up with it because I get 

`POST /api/form-state 401 in 7ms
[16:12:20] ERROR: Forbidden: You are not allowed to perform this action.`

anytime I try to edit the page or post collection. It should work out of the box

**Reply by denolfe** [reply]
Would need more info and a recreation. Please open an issue.

**Reply by klauss194** [reply]
make sure the sqlite file has writing rights ?
